### PR TITLE
Update name error message

### PR DESCRIPTION
**Pull request by tonishka**

### CODE REVIEW

**Review by ckcherry23 — APPROVED** [review]

File: src/main/java/seedu/address/model/person/Name.java [hunk 1]
```diff
@@ -10,7 +10,9 @@
 public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain English alphabetic characters, special characters . (period) ' (apostrophe) and " +
```
Comment (ckcherry23) [inline src/main/java/seedu/address/model/person/Name.java:13]:
```suggestion
            "Names should only contain English alphabetic characters, special characters . (period) ' (apostrophe) " +
```

File: src/main/java/seedu/address/model/person/Name.java [hunk 2]
```diff
@@ -10,7 +10,9 @@
 public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Names should only contain alphanumeric characters and spaces, and it should not be blank";
+            "Names should only contain English alphabetic characters, special characters . (period) ' (apostrophe) " +
+                    "- (dash) and spaces, " +
+                    "and it should not be blank";
```
Comment (ckcherry23) [inline src/main/java/seedu/address/model/person/Name.java:15]:
```suggestion
            "Names should only contain English alphabetic characters, special characters . (period) "
                    + "' (apostrophe) - (dash) and spaces, "
                    + "and it should not be blank";
```

Comment (ckcherry23) [reply src/main/java/seedu/address/model/person/Name.java:15]: There's a checkstyle main issue

Comment (tonishka) [reply src/main/java/seedu/address/model/person/Name.java]: I fixed checkstyle just now

**Pull request merged**